operator-pipeline-images/operatorcert/webhook_dispatcher/config.py

```diff
@@ -60,6 +62,77 @@ class CapacityConfig(BaseModel):
     namespace: str
 
 
+class Filter(BaseModel):
+    """Filter configuration for webhook events."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    cel_expression: str = Field(
+        default="",
+        description="CEL expression to filter events. "
+        "If empty, no filtering is applied.",
+    )
+
+    @field_validator("cel_expression")
+    @classmethod
+    def validate_cel_expression_and_compile(cls, value: str) -> str:
+        """
+        Validate the CEL expression syntax.
+
+        Raise ValueError if the expression is invalid.
+
+        Args:
+            value (str): A raw CEL expression value.
+        """
+        if not value:
+            return value
+
+        cls.compile_expression(value)
+
+        return value
```

Comment:
Is there any reason not to do something like this?

```suggestion
    cel_expression: lark.Tree = Field(
        default=cls.compile_expression("true"),
        description="CEL expression to filter events. "
        "If empty, no filtering is applied.",
    )

    @field_validator("cel_expression")
    @classmethod
    def validate_cel_expression_and_compile(cls, value: str) -> lark.Tree:
        if not value:
            return cls.compile_expression("true")

        return cls.compile_expression(value)
```

Reply:
Apart from that suggested code always initializes and compiles the expression even if it is not present in the config. I would say this is just a different implementation. And I would like to avoid evaluating the expression in the dispatcher in case filters are optional.